src/test/java/Drivers/WebDriverFactory.java

```diff
@@ -0,0 +1,10 @@
+package Drivers;
+
+import org.openqa.selenium.WebDriver;
+
+import java.net.MalformedURLException;
+
+public abstract class WebDriverFactory {
+    public abstract WebDriver initializeDriver() throws MalformedURLException;
```

Comment:
Why do you need to throw this exception? On my mind you must handle it in implementations, because if URL is incorrect, then it doesn't make sense to continue test execution